Map/Include/Map.hpp

```diff
@@ -1,13 +1,15 @@
 #pragma once
-#include "Room.hpp"
+#include "Field.hpp"
+#include <memory>
 #include <vector>
 
+using FieldList = std::vector<std::vector<std::unique_ptr<Field>>>;
+
 class Map
 {
 public:
-    std::vector<Room> getRooms();
-    Map();
-    Map(int mapSize);
-private:
-    std::vector<Room> rooms;
+    virtual FieldList const& getFields() = 0;
+
+protected:
```

Comment:
To do private w klasie pochodnej. Na ten moment nie ma żadnych wskazań do tego, żeby miało to być tutaj protected